pkg/client/core/api/git/nameserverdelegator_api.go

```diff
@@ -0,0 +1,282 @@
+// Package git knows how to do git operations
+package git
+
+import (
+	"fmt"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/go-git/go-billy/v5/memfs"
+
+	"github.com/go-git/go-billy/v5"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/go-git/go-git/v5/storage/memory"
+	"github.com/oslokommune/okctl/pkg/client"
+	okctlconfig "github.com/oslokommune/okctl/pkg/config"
+	"github.com/oslokommune/okctl/pkg/github"
+)
+
+type repository struct {
+	Endpoint       string
+	Organization   string
+	RepositoryName string
+}
+
+func (r repository) String() string {
+	return fmt.Sprintf("%s:%s/%s.git", r.Endpoint, r.Organization, r.RepositoryName)
+}
+
+// NameserverDelegationRequest represents a single request for a nameserver record
+type NameserverDelegationRequest struct {
+	githubAPI github.Githuber
+
+	Record *client.NameserverRecord
+}
+
+// Submit creates a pull request to the IAC repo with the NS record
+func (n *NameserverDelegationRequest) Submit() (err error) {
+	err = n.githubAPI.CreateNSRecordPullRequest(n.Name())
+	if err != nil {
+		return fmt.Errorf("error creating pull request: %w", err)
+	}
+
+	return nil
+}
+
+// Name returns a value that can be used to identify a request
+func (n *NameserverDelegationRequest) Name() string {
+	return strings.Split(n.Record.FQDN, ".")[0]
+}
+
+// IsSubmitted checks if theres already a PR for the relevant NS record
+func (n *NameserverDelegationRequest) IsSubmitted() bool {
+	hasExisting, err := n.githubAPI.HasExistingNSRecordPullRequest(n.Name())
+	if err != nil {
+		return false
+	}
+
+	return hasExisting
+}
+
+// NewNameserverDelegationRequest instantiates a DelegationRequest struct
+func NewNameserverDelegationRequest(gh github.Githuber, record *client.NameserverRecord) *NameserverDelegationRequest {
+	return &NameserverDelegationRequest{
+		githubAPI: gh,
+		Record:    record,
+	}
+}
+
+// NameserverDelegator handles functionality regarding delegation of NS records
```

Comment:
I would like to somewhere be able to read what a "nameserverDelegationReconciler" really is (for instance "nameserverDelegationReconciler waits for a specified hosted zone to exist before continuing", or something more correct). Shouldn't that be here?

(I see this is an issue for all reconcilers. I also realize now that I'm missing a documentation of what a reconciler is in reconciler.go, but that's out of scope for this PR.)